scripts/globals/spells/retrace.lua

```diff
@@ -0,0 +1,28 @@
+-----------------------------------------
+-- Spell: Retrace
+-- Transports player to their Allied Nation. Can cast on allies.
+-----------------------------------------
+require("scripts/globals/teleports")
+require("scripts/globals/status")
+require("scripts/globals/msg")
+-----------------------------------------
+
+function onMagicCastingCheck(caster, target, spell)
+    local result = 0
+    if (target:getCampaignAllegiance() > 0) then
+        result = 0
+    else
+        result = 48
+    end
+    return result
+end
+
+function onSpellCast(caster, target, spell)
+    if (target:getCampaignAllegiance() > 0) then
+        target:addStatusEffectEx(tpz.effect.TELEPORT, 0, tpz.teleport.id.RETRACE, 0, 3.4)
```

Comment:
3.4 isn't actually valid because the function doesn't support floating point values. it will become just 3 at that point. I am aware we have some existing code making the same mistake that has not been corrected yet.

Reply:
I used Warp II as a reference since this spell also is cast on other people, it appears 4 is more correct though yeah. Wasn't aware of that, thanks.